src/sql/arrow_sql_gen/sqlite.rs

```diff
@@ -130,6 +124,32 @@ pub fn rows_to_arrow(mut rows: Rows, num_cols: usize, projected_schema: Option<S
     }
 }
 
+fn to_sqlite_decoding_type(data_type: &DataType) -> DataType {
+    match data_type {
+        DataType::Null => DataType::Null,
+        DataType::Int8 => DataType::Int8,
+        DataType::Int16 => DataType::Int16,
+        DataType::Int32 => DataType::Int32,
+        DataType::Int64 => DataType::Int64,
+        DataType::UInt8 => DataType::UInt8,
+        DataType::UInt16 => DataType::UInt16,
+        DataType::UInt32 => DataType::UInt32,
+        DataType::UInt64 => DataType::UInt64,
+        DataType::Boolean => DataType::Boolean,
+        DataType::Float16 => DataType::Float16,
+        DataType::Float32 => DataType::Float32,
+        DataType::Float64 => DataType::Float64,
+        DataType::Utf8 => DataType::Utf8,
+        DataType::LargeUtf8 => DataType::LargeUtf8,
+        DataType::Binary | DataType::LargeBinary | DataType::FixedSizeBinary(_) => DataType::Binary,
+        DataType::Decimal128(_, _) | DataType::Decimal256(_, _) => DataType::Float64,
```

Comment:
This is existing behavior. Will be improved to directly convert to Decimal by separate PR